### ISSUE TITLE
GIT_REFLOW_PR_MSG: Permission denied

### ISSUE THREAD

**Issue by sethladd** [reenhanced/gitreflow#115]
After running git reflow review on my first branch created by reflow, I saw this error:

```
~/Code/dartdoc[sl-hide-exceptions] $ git reflow review
sh: /Users/sethladd/Code/dartdoc/.git/GIT_REFLOW_PR_MSG: Permission denied
```

It looks like I'm not able to edit my PR message, and maybe this error is why?

**Comment by codenamev** [comment]
We've been getting this lately from others as well... can you do a quick `ls -la /Users/sethladd/Code/dartdoc/.git` and share the output?  I'm using `zsh` and haven't had these problems, but for those that use `bash` we're seeing these errors.  I'll investigate soon and get this resolved for you.  In the meantime, a dirty workaround would be to `touch .git/GIT_REFLOW_PR_MSG` before you review.

**Comment by sethladd** [comment]
```~/Code/dartdoc[master] $ ls -la .git
total 72
drwxr-xr-x   16 sethladd  eng   544 Jul 24 08:02 .
drwxr-xr-x   29 sethladd  eng   986 Jul 23 16:27 ..
-rw-r--r--    1 sethladd  eng    46 Jul 23 16:47 COMMIT_EDITMSG
-rw-r--r--    1 sethladd  eng   204 Jul 23 16:52 FETCH_HEAD
-rw-r--r--    1 sethladd  eng    23 Jul 23 16:52 HEAD
-rw-r--r--    1 sethladd  eng    41 Jul 23 16:37 ORIG_HEAD
drwxr-xr-x    2 sethladd  eng    68 May 18 16:02 branches
-rwxr--r--    1 sethladd  eng  1369 Jul 23 16:52 config
-rw-r--r--    1 sethladd  eng    73 May 18 16:02 description
drwxr-xr-x   12 sethladd  eng   408 Jul 21 13:17 hooks
-rw-r--r--    1 sethladd  eng  8072 Jul 23 16:52 index
drwxr-xr-x    3 sethladd  eng   102 May 18 16:02 info
drwxr-xr-x    4 sethladd  eng   136 May 18 16:02 logs
drwxr-xr-x  260 sethladd  eng  8840 Jul 16 16:14 objects
-rw-r--r--    1 sethladd  eng   508 May 18 16:02 packed-refs
drwxr-xr-x    5 sethladd  eng   170 May 29 14:11 refs
```

**Comment by codenamev** [comment]
I was worried about that... looks like we might have to hijack the `COMMIT_EDITMSG`  I'll try to get to this today

**Comment by sethladd** [comment]
Thanks for taking a look!

**Comment by sethladd** [comment]
FWIW, looks like touching that file doesn't help:

```
~/Code/dartdoc/test_package[sl-dual-nav] $ touch /Users/sethladd/Code/dartdoc/.git/GIT_REFLOW_PR_MSG
~/Code/dartdoc/test_package[sl-dual-nav] $ git reflow review
sh: /Users/sethladd/Code/dartdoc/.git/GIT_REFLOW_PR_MSG: Permission denied

Review your PR:
--------
Title:
sl-dual-nav

Body:

--------
Submit pull request? (Y)
^Cerror: 

/usr/bin/git-reflow: Interrupt
```

**Comment by sethladd** [comment]
Any thoughts on this? We'd like to roll this out to more of our team, but everyone is using bash. Thanks!

**Comment by codenamev** [comment]
I should have time to address this tomorrow.   I have a hunch it's the type of ruby system call that it makes, and if not we'll have to resort to storing it in the gitconfig.

**Comment by sethladd** [comment]
Hi,

Just tried this, but now I'm getting:

```
~/Code/dartdoc[sl-test-resources] $ git reflow review
error: undefined method `empty?' for nil:NilClass
```

This is the first review for a new branch.

**Comment by codenamev** [comment]
Hey @sethladd,

This error surrounds not setting of your `EDITOR` environment variable.  For now, if you set it you shouldn't have an issue.  I'll look into getting this fixed possibly by end of day/beginning tomorrow.

**Comment by sethladd** [comment]
Gotcha! I set `EDITOR` and sure enough, it works!

**Comment by codenamev** [comment]
@sethladd f.y.i. fixed in `0.6.3` :-)

**Comment by sethladd** [comment]
:+1:
